azurerm/resource_arm_redis_cache.go

```diff
@@ -406,19 +426,22 @@ func resourceArmRedisCacheRead(d *schema.ResourceData, meta interface{}) error {
 		d.Set("location", azureRMNormalizeLocation(*location))
 	}
 
-	d.Set("ssl_port", resp.SslPort)
-	d.Set("hostname", resp.HostName)
-	d.Set("port", resp.Port)
-	d.Set("enable_non_ssl_port", resp.EnableNonSslPort)
-
 	if sku := resp.Sku; sku != nil {
 		d.Set("capacity", sku.Capacity)
 		d.Set("family", sku.Family)
 		d.Set("sku_name", sku.Name)
 	}
 
-	if resp.ShardCount != nil {
-		d.Set("shard_count", resp.ShardCount)
+	if props := resp.Properties; props != nil {
+		d.Set("ssl_port", props.SslPort)
```

Comment:
Shall we build a helper method to flatten it ?

Reply:
nope - we only do that for complex (e.g. nested) field, a single field isn't necessary